### ISSUE TITLE
[Bug]: GMocked class cannot be used in std::tuple

### ISSUE THREAD

**Issue by derived-coder** [google/googletest#4274]
### Describe the issue

I got following compiler error when using following code:
```
#include "gmock/gmock.h"
#include "gtest/gtest.h"
#include <string>
#include <tuple>

using namespace ::testing;


class MyMock {
 public:
    explicit MyMock(std::string foo) {}
  
   MOCK_METHOD(void, Push, (const int& x)){}; // when I uncomment this code, the code compile
};


TEST(SutTest, UseMock) {
    
    std::tuple<MyMock> foo{MyMock(std::string{})};
}

```
See following working code: https://godbolt.org/z/5j1eYb4da

### Steps to reproduce the problem

see the code snippet

### What version of GoogleTest are you using?

1.10.0

### What operating system and version are you using?

linux

### What compiler and version are you using?

> gcc8

### What build system are you using?

cmake

**Comment by pgawro** [comment]
Hi @derived-coder  

I think instead of using   `    std::tuple<MyMock> foo{MyMock(std::string{})};
`
use ` std::tuple<MyMock> foo("");` that should help for your compilation problem.
Let me know if that helped

**Comment by derived-coder** [comment]
@pgawro 
you are right, this works, but it is more complicated. It will not work, when I have more then one argument in the ctor.
The problem is that a GMock object is made implicit non-copyable.

Is it possible to workaround this?

I have now following case where I need to return a copy of a mock object:
``` 
static MyMock setup()
{
    return MyMock();
}
```
https://godbolt.org/z/vrd49Phae

I can tell you what I am actually doing: 
```
template<typename T>
struct MyClassUnderTest
{
std::array<T, 10> m_dep;
};

// test code:
MyClassUnderTest<MyMock> t;
// t...
```
